R/utils-get_code_dependency.R

```diff
@@ -380,8 +380,12 @@ extract_side_effects <- function(calls_pd) {
 #' @keywords internal
 #' @noRd
 graph_parser <- function(x, graph) {
+  # normalize x to remove surrounding backticks
+  x <- gsub("^`|`$", "", x)
+  graph <- lapply(graph, function(call) gsub("^`|`$", "", call))
```

Comment:
I think normalization here is not needed. We can base on the syntactic symbols containing ticks. This would require to pass `x` as name(s).

Reply:
We need normalization with current codebase, first because it's not possible to convert with R's API to convert string with backticks into a symbol without backticks (see reproducible code)

``` r
x <- "add column"
as.name("add column") == "`add column`"
#> [1] FALSE
identical(as.name("add column"), "`add column`")
#> [1] FALSE

as.name("`add column`")
#> `\`add column\``
as.name("add column") == as.name("`add column`")
#> [1] FALSE
identical(as.name("add column"), as.name("`add column`"))
#> [1] FALSE
```

Second, the normalization is necessary due to an edge case seen in the image below, where the `graph` has both backticked and non-backticked elements.

This makes it impossible to compare using the existing logic

![image](https://github.com/user-attachments/assets/578fe4ad-fb55-42de-a51c-681b1e834d8e)

**My suggestion**: Looking back it is wiser to to perform the normalization outside the code parser, in `code_graph`'s body (or in the functions that it call)

Reply:
See 7ca9deeb